### PR TITLE
Added unittest assertion checker -- BOM-2339

### PR DESCRIPTION
**Pull request by iamsobanjaved**
JIRA:  https://openedx.atlassian.net/browse/BOM-2339

### CODE REVIEW

File: edx_lint/pylint/unittest_assert_check.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        "assertIsNone": "assert arg1 is None",
+        "assertIsNotNone": "assert arg1 is not None",
+        "assertIsInstance": "assert isinstance(arg1, arg2)",
+        "assertNotIsInstance": "assert not isinstance(arg1, arg2)",
+        "assertLess": "assert arg1 < arg2",
+        "assertLessEqual": "assert arg1 <= arg2",
+        "assertGreater": "assert arg1 > arg2",
+        "assertGreaterEqual": "assert arg1 >= arg2",
+        "assertAlmostEqual": "assert round(arg1 - arg2, 7) >= 0",
+        "assertNotAlmostEqual": "assert round(arg1 - arg2, 7) != 0",
```
Comment (nedbat) [inline edx_lint/pylint/unittest_assert_check.py:72]: The math module has [isclose](https://docs.python.org/3/library/math.html#math.isclose) which would be a better replacement.

File: test/plugins/test_unittest_assert_check.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+import pytest
+
+from .pylint_test import run_pylint
+
+
+def test_bad_asserts():
+    source = """\
+        import unittest
+
```
Comment (mraarif) [inline test/plugins/test_unittest_assert_check.py:11]: maybe move this to a `.py` file and provide a path to that file instead

Comment (iamsobanjaved) [reply test/plugins/test_unittest_assert_check.py:11]: This is the trend in entire `edx-lint/test/plugins/` directory.

https://github.com/edx/edx-lint/tree/38a1b8580e8621629422cf7fd57d7399c9587bf0/test/plugins

Comment (mraarif) [reply test/plugins/test_unittest_assert_check.py:11]: it shouldn't be hard to read from a file instead, it'd be a lot cleaner solution

File: edx_lint/pylint/unittest_assert_check.py (hunk 1 above)
Comment (iamsobanjaved) [reply edx_lint/pylint/unittest_assert_check.py]: Replaced with `isclose`

File: test/plugins/test_unittest_assert_check.py (hunk 2 above)
Comment (nedbat) [reply test/plugins/test_unittest_assert_check.py]: An advantage of keeping the code in a string is that I can read and understand the whole test without having to go find another file.

Comment (iamsobanjaved) [reply test/plugins/test_unittest_assert_check.py:11]: I have moved the test code back to strings

**Review by jmbowman — APPROVED** [review]
One wording improvement suggestion, otherwise looks good to me.

**Pull request merged**